R/modules.R

```diff
@@ -321,11 +321,124 @@ modules <- function(..., label = "root") {
 # printing methods ----
 
 #' @rdname teal_modules
+#' @param what character vector specifying which metadata to display. 
+#'   Possible values: "data_sources", "properties", "ui_args", "server_args", "transformers"
 #' @export
-format.teal_module <- function(x, indent = 0, ...) {
-  paste0(paste(rep(" ", indent), collapse = ""), "+ ", x$label, "\n", collapse = "")
+format.teal_module <- function(x, indent = 0, is_last = FALSE, parent_prefix = "", what = c("data_sources", "properties", "ui_args", "server_args", "transformers"), ...) {
+  empty_text <- "─ none"
+  branch <- if (is_last) "└─" else "├─"
+  current_prefix <- paste0(parent_prefix, branch, " ")
+  content_prefix <- paste0(parent_prefix, if (is_last) "   " else "│  ")
+
+  format_list <- function(lst, empty = empty_text) {
+    if (is.null(lst) || length(lst) == 0) {
+      empty
+    } else {
+      paste(names(lst), collapse = ", ")
+    }
+  }
+
```

Comment:
Hey @vedhav wanted to talk about the depth and the width. Did you consider a parameter that controls the depth of the print? Maybe let's print up to 3rd depth by default and we could trim lines at 80 or 120 or 240 digits/characters?


Depth could control if we actually print the module marked with blue, and the width could control how long is the line with the output (marked with red)?

<img width="445" alt="image" src="https://github.com/user-attachments/assets/dd3da465-8421-4cdf-8388-97c237fed3eb">

Reply:
This is just a nice to have for the people who like to print it on a small area vs those who wants to print EVERYTHING : P

Reply:
I tried to achive this but it was non-trivial. It made the code much complex. I'll think about this some more.

Reply:
Resolving this since we went with the other suggestion to keep them vertical instead.